src/js/main.js

```diff
@@ -47,30 +48,144 @@ Site = {
 };
 
 Site.Overlay = {
+  activeNavItem: null,
+  desktopWidth: 1024,
+
   init: function() {
     var _this = this;
 
+    // GET! ALL THE THINGS!
+    _this.pageOverlay = document.getElementsByClassName('page-overlay');
+
+    _this.header = document.getElementById('header');
+
     _this.mobileToggle = document.getElementById('mobile-overlay-toggle');
     _this.mobileMenu = document.getElementById('mobile-menu');
 
-    _this.bindToggles();
+    _this.pageOpen = document.getElementsByClassName('js-page-open');
+
+    _this.desktopPageClose = document.getElementsByClassName('desktop-page-close');
+
+    // BIND CLICKS!
+    _this.bindClicks();
   },
 
-  bindToggles: function() {
+  bindClicks: function() {
     var _this = this;
 
-    _this.mobileToggle.addEventListener('click', _this.handleToggle.bind(_this));
+    // bind the mobile toggle clicks!
+    _this.mobileToggle.addEventListener('click', _this.toggleMobile.bind(_this));
+
+    for (var i = 0; i < _this.pageOpen.length; i++) {
+      // bind the page nav item clicks!
+      _this.pageOpen[i].addEventListener('click', _this.openPage.bind(_this));
+    }
+
+    for (var i = 0; i < _this.desktopPageClose.length; i++) {
+      // bind the desktop page close X clicks!
+      _this.desktopPageClose[i].addEventListener('click', _this.closePage.bind(_this));
+    }
   },
 
-  handleToggle: function() {
+  toggleOverlay: function() {
     var _this = this;
 
     if (document.body.classList.contains('overlay-active')) {
+      // if the overlay is showing: hide it
       document.body.classList.remove('overlay-active');
     } else {
+      // otherwise: show it
       document.body.classList.add('overlay-active');
     }
 
+  },
+
+  toggleMobile: function() {
+    var _this = this;
+
+    // on mobile, the same button closes pages,
+    // and also toggles the mobile menu
+
+    // if there is a page open
+    if (document.querySelector('.page-active') !== null) {
+      // close it
+      _this.closePage();
+      // and hide the overlay
+      _this.toggleOverlay();
+
+    // otherwise, we're toggling the menu
+    } else {
+
+      // if the menu is open
+      if (_this.mobileMenu.classList.contains('active')) {
+        // close it
+        _this.mobileMenu.classList.remove('active');
+      } else {
+        // otherwise, open it
+        _this.mobileMenu.classList.add('active');
+      }
+    }
+
+    // and finally, hide or show the overlay
+    _this.toggleOverlay();
+  },
+
+  openPage: function(event) {
+    var _this = this;
+
+    // get the page name from the
+    // clicked <a> data-page attribute
+    var pageName = event.target.getAttribute('data-page');
+
+    // find the page container with that ID
+    _this.activePage = document.getElementById('page-' + pageName);
+    // open the page!
+    _this.activePage.classList.add('page-active');
+
+    // set the <li> parent of the target <a>
```

Comment:
can't use the `<li>` in the desktop nav tho, because it's the parent for both the link and the X. so we have to differentiate those somehow